\chapter{Introduction}
Genetic novelty, a crucial property of all biological systems, originates as mutation. Mutation is the coupling of two components: (1) lesion formation and (2) the subsequent failure of DNA repair systems to recapitulate the original sequence. Through observations of sequence composition alone, it is apparent that the dynamics of this process vary both between species \citep{Karlin1994ComparisonsSequences} and within genomes \citep{Francioli2015Genome-wideHumans}. Although this implies that changes in mutagenesis are a feature of the evolution of DNA sequences, nearly all statistical models in molecular evolution and phylogenetics assume that this property does not exist. An understanding of the properties of mutagenesis is key to be able to robustly identify when natural selection is operating. Although compositional differences imply historical changes in mutation, it does not guarantee that this mutation disequilibrium persists. Accordingly, methods that purport to tackle the existence of mutation disequilibrium have been developed \citep{Squartini2008QuantifyingProcess, Singh2009StrongDrosophila, Ababneh2006Matched-pairsSequences}. Here, I outline the problems with the existing methods and propose new methods for detecting this property. I demonstrate that the new methods are fit for purpose with a complementary experimental design, coupling simulation studies with analyses of empirical data known to have been subjected to the process that I am seeking to detect. I focus on evaluating whether the new methods, when applied to such empirical cases, are coherent with the biological basis.

As mutation is the raw material of evolution, without a fuller understanding of mutagenesis, a mechanistic understanding of evolution is incomplete. On a chemical level, DNA is the same for all species. Yet phenotypic differences are extensive, reflecting the variation in the organisation of DNA into a genome. Pre-genomics era analyses observed that genomes could be distinguished by their `general design', attributing this variation to differences in the biochemistry associated with the management of DNA \citep{Karlin1994ComparisonsSequences, Karlin1995DinucleotideSignature}. The neutral theory of molecular evolution, proposing that most genetic variation has not been affected by natural selection but instead reflects the random chance of mutation events \citep{Kimura1968EvolutionaryLevel, King1969Non-DarwinianEvolution}, provides a useful null hypothesis to evolution. Accordingly, our understanding of the null is dependent on our understanding of the full scope neutral mutagenic processes. Species-specific mechanisms of lesion formation (cite) and DNA repair (cite) imply preceding periods of disequilibrium of mutation. Determining the existence and impact of such changes in mutagenesis is therefore required to robustly reject the null and identify when natural selection is operating.

The mutational process is not immune to the forces of evolution \citep{Lynch2016GeneticRate}. Mechanisms of mutagenesis, such as DNA polymerase, DNA repair proteins, or DNA methylase are encoded within the genome and as a consequence, are subject to forces of evolution \citep{Lynch2010EvolutionRate}. The loss of a pronounced number of genes associated with the cell-cycle and DNA repair was identified in a group of budding yeasts \citep{Steenwyk2019ExtensiveYeasts}. This loss was shown to be accompanied by an increase in mutational load, change in the composition of mutations, and an increase in the rate of evolution \citep{Steenwyk2019ExtensiveYeasts}. Such a genetic change that modifies either the rate or composition of mutagenesis is defined as a mutator (or antimutator) allele \citep{Lynch2016GeneticRate}.

As the mutagenic process appears to vary both within and between genomes, mutation disequilibrium must be considered at such local and global scales. This variation can be illustrated simply with sequence composition as an indirect measure of the processes acting upon the genome. Distinct compositions span the tree of life, for example, the percent of nucleotide bases which are either guanine or cytosine (GC content) of \textit{Plasmodium falciparum} is 24\% while \textit{Mycobacterium tuberculosis} is  66\% \citep{Nakamura2000Codon2000}. This implies changes in mutagenesis on a global scale. However, composition differences within genomes is also a long-standing observation. Vertebrate genomes are described as a mosaic of isochores of alternating low and high GC contents \citep{Bernardi1989TheGenome, Bernardi2000IsochoresVertebrates}. This implies local differences in biochemistry and/or mutation. 

DNA methylation strongly affects the mutational profile of a species, and accordingly the deletion of the genetic sequence that encodes it is an example of a global evolution of a mutator. In comparison to non-methylated cytosine, 5-methylcytosine ($^5$mC) is hypermutable, with higher rates of deamination to thymine \citep{Coulondre1978MolecularColi}. DNA methylation predominantly occurs on cytosines that precede a guanine nucleotide, known as CpG dinucleotides \citep{Holliday1975DNADevelopment}. CpG methylation has been identified in the genomes of a diverse range of invertebrates, including insects \citep{Wang2010EstimatingLoci}. However, the current evidence supports only trace levels of cytosine methylation in \textit{Drosophila melanogaster} \citep{Capuano2014CytosineSpecies, Deshmukh2018LevelsGenome}. The methyltransferases DNMT1 and DNMT3, reported to be necessary for a functional methylation system are not present in the \textit{D. melanogaster} genome, which possesses a sole DNMT2 methyltransferase \citep{Goll2005EukaryoticMethyltransferases, Tweedie1999VestigesMelanogaster}. Low levels of DNA methylation are a feature of the order Diptera \citep{Bewick2017EvolutionInsects, Provataris2018SignaturesHolometabola}, yet \textit{D. melanogaster} stands out with distinctly low levels reduced by well over an order of magnitude \citep{Deshmukh2018LevelsGenome}. The level of methylation in \textit{D. melanogaster} is 50 times lower than that in \textit{D. simulans} \citep{Deshmukh2018LevelsGenome}. These observations suggest that following divergence, generally estimated to be between 0.8 and 5.4 MYA \citep{Cutter2008DivergenceRate, Wang2010EstimatingLoci, Tamura2004TemporalClocks}, DNA methylation has been close to lost in \textit{D. melanogaster}. This leads to the prediction that the \textit{D. melanogaster} genome, having experienced a recent evolution of a mutator, would globally have a higher level of mutation disequilibrium than \textit{D. simulans}. 

Some genomic features can also have a localised effect on mutation \citep{Lynch2016GeneticRate}. Long stretches of DNA that are homogeneous in base composition, known as isochores, are a feature of vertebrate genomes \citep{Bernardi1989TheGenome, Bernardi2000IsochoresVertebrates}. A high recombination rate has been associated with a high rate of mutation to create GC base pairs, implicating recombination as a determinant of isochore organisation \citep{Montoya-Burgos2003RecombinationGenomes}. In \textit{Mus musculus}, \textit{Fxy}, a transcribed gene, spans the boundary of the pseudo-autosomal region (PAR) \citep{Palmer1997AMice}. The PAR is a region of homology between the X and Y chromosomes in mammals, undergoing one obligatory crossover per generation.  In other \textit{Mus} taxa, \textit{Fxy} is X-specific, suggesting translocation to the PAR in \textit{M. musculus} after divergence with \textit{M. spretus}, between 2-3 million years ago \citep{Huang2005HowMammals}. The half of \textit{Fxy} located in the PAR (frequent recombination) exhibits differing rates of evolution to the other X-specific half (no recombination) \citep{Perry1999EvolutionaryPosition}. This is an interesting natural experiment involving a genomic sequence putatively subject to a new mutagenic environment. There is a strong expectation that there would be a local elevation of mutation disequilibrium within the gene, in the PAR-located half. 

The rate of mutation varies greatly between species, and even between sequence type (ref). The ultimate end state of genetic systems appears to be a balance between mutation and selection. It is theorised that selection acts to suppress mutator alleles indirectly by operating on the deleterious changes made by a mutator elsewhere in the genome \citep{Lynch2010EvolutionRate}. The lower bound of the mutation rate is set by the power of random genetic drift which, in turn is determined by the effective population size \citep{Lynch2010EvolutionRate}. 

The relative levels of purifying natural selection operating on a genomic segment allows for predictions of the magnitude of mutation disequilibrium. For hetero-gametic sexes, you expect the chromosome that is hemizygous to be subjected to more stringent natural selection, because any recessive deleterious gene is exposed in the homozygous sex (cite). The prediction is that with the increased magnitude of purifying selection, the rate of convergence to equilibrium should be slower. A slower rate of convergence to equilibrium leads to a higher magnitude of disequilibrium. Accordingly, I expect the chromosome that is hemizygous to exhibit a higher magnitude of disequilibrium relative to the autosomes. The demonstrated abundance of functional elements in the genome outside of protein coding sequence is very low \citep{Graur2013OnENCODE}. Accordingly, changes in intronic sequences are unlikely to be detrimental, making introns a useful neutral benchmark (until proven otherwise). Consequently, I expect CDS sequences to exhibit a higher magnitude of mutation disequilibrium than intronic sequences that it flanks. 

Probabilistic models of sequence divergence, known as substitution models, provide a basis for statistical inference of the evolutionary process. Substitution models use a Markov process to describe how evolution happened on a branch of a phylogenetic tree. The primary approach to gaining inference from substitution models is through a Maximum Likelihood (ML) framework. ML will be used in this work for its established theory and associated properties. Early substitution models were simple, with assumptions relaxed over time, however, certain assumptions have persisted through to the models currently in use. The prevailing assumptions are: the evolutionary process remains the same over time (time-homogeneity); the process is identical to how it would appear if run in reverse (reversibility); and the process is in \gls{equilibrium} (stationarity). It is worth noting that reversibility implies stationarity, but the reverse is not true. Such assumptions are made for mathematical simplicity and conflict with our understanding of the true biological process. 

Given the evidence that the processes affecting mutagenesis can change through time, defying the assumption of equilibrium, tests of disequilibrium have been proposed before. \cite{Ababneh2006Matched-pairsSequences} introduced matched-pairs tests of homogeneity, looking at the internal and marginal symmetry of the nucleotide composition. These methods, however, are limited in power because they operate on pairs of sequences. \cite{Singh2009StrongDrosophila} introduced a time to equilibrium calculation of the GC content of a single edge, denoted $t_{eq}$. However, fitting a general but strand-symmetric process to determine the equilibrium distribution, $t_{eq}$ is limited to such processes. Furthermore, because $t_{eq}$ is purely in terms of GC content, it does not capture the complete scope of possible change. \cite{Squartini2008QuantifyingProcess} introduced three indices to describe stationarity, calculated from nucleotide composition, together denoted STIs. Combining the STIs to address questions about equivalence requires ad hoc approaches. The STIs are not tests of non-stationarity, instead, they use a $\chi^2$-test comparing the current and equilibrium nucleotide distributions to determine significance. The test proposed by \cite{Squartini2008QuantifyingProcess} appears to have a good theoretical basis, however, they do not evaluate whether the test statistic is consistent with the asymptotic approximations. Furthermore, they do not demonstrate that it behaves in a reliable manner. It remains unclear as to whether the methods are, in fact, reasonable. 

Essential to developing and validating methods to detect mutation disequilibrium is a paired experimental design.The essential components to developing methods to detect mutation disequilibrium are appropriate positive and negative controls. This requires defining an appropriate null hypothesis and using it to generate data meaningfully, to construct edge cases with simulated data that establish the consistency of the methods with theoretical expectations. Such simulations create data in accordance with models, necessary to understand a statistic's behaviour where the properties of the data are known. However, simulations alone are not enough, given the data are generated by a model, and not by nature. The ultimate benchmark is to take methods to cases where empirical evidence clearly indicates the presence of a mutator, and so there is strong evidence for the existence of mutation disequilibrium. Having such a positive empirical control is essential for confidence in the methods. The failure to carry out these steps is the major limitations of the previous work in the field. 

To this end, I use the \textit{D. melanogaster} genome and the \textit{Fxy} gene in \textit{M. musculus} as known cases of the presence of a mutator. My overarching aim is to develop methods that elucidate the status of disequilibrium in DNA sequences. I seek to establish whether I can detect mutation disequilibrium on a single edge, robustly measure its magnitude, and establish if two mutational processes are the same. I investigate whether, with my developed methods, I can detect elevated mutation disequilibrium in \textit{D. melanogaster} compared to \textit{D. simulans}, and in the half of \textit{Fxy} in the PAR in \textit{M. musculus}. After trialling my methods on these two examples, I then explore whether I can determine if the human genome is at equilibrium.

In this work, I develop Likelihood Ratio Tests for: (1) determining the existence of mutation disequilibrium; and (2) the equivalence between mutagenic processes. I present a statistic to quantify the level of disequilibrium, denoted $\nabla$, and demonstrate that it exhibits robust behaviour. The vast majority of the \textit{D. melanogaster} genome was not estimated to be in mutation disequilibrium, whereas the extent of mutation disequilibrium in sister was taxa substantially lesser, even in terms of measures of magnitude. The predictions for \textit{Fxy} largely held also, showing elevated disequilibrium in the region that has moved into the PAR relative to the component of the gene that remains in the uniquely X region. These findings show that application of my new methods to data sets with strong prior empirical evidence of changes to mutagenesis gives results that are consistent with expectations. This allows for confidence in the estimate given in application to the human genome, for which I argue that the majority of sequence is not in mutation equilibrium. 