\makeglossaries

\newglossaryentry{floating point arithmetic}
{
        name=Floating Point Arithmetic,
        text=floating point arithmetic,
        description={the arithmetic performed on floating-point representations of numbers by computers and other automated devices}
}


\newglossaryentry{branch length}
{
        name=Branch Length,
        text=branch length,
        description={ a measure of genetic distance between taxa on a phylogenetic tree, the expected number of substitutions per site for a specific branch}
}


\newglossaryentry{strand-asymmetric}
{
        name=Strand-Asymmetry,
        text={strand-asymmetric},
        description={the differential distribution of nucleotides between the two DNA strands}
}

\newglossaryentry{strand-symmetric}
{
        name=Strand-Symmetry,
        text=strand-symmetric,
        description={when numbers of occurrence of nucleotides match those of their respective complements in a DNA strand of sufficient length, i.e., (A = T and C = G)}
}


\newglossaryentry{equilibrium}
{
        name=Equilibrium,
        text=equilibrium,
        description={the state where the population nucleotide frequencies of a DNA sequence remain unchanged through time}
}

\newglossaryentry{stationarity}
{
        name=Stationarity,
        text=stationarity,
        description={when a substitution model is in a state of the \gls{equilibrium}}
}


\newglossaryentry{edge}
{
        name=Edge,
        text=edge,
        description={(of a tree) the evolutionary transition from an ancestral sequence to a descendant sequence}
}

\newglossaryentry{consistency}
{
        name=Consistency,
        text=consistency,
        description={The theoretical possibility of learning the true values of this model's underlying parameters after obtaining an infinite number of observations}
}

\newglossaryentry{identifiable}
{
        name=Identifiable,
        text=identifiable,
        description={A model is identifiable if the probability distribution of observation uniquely corresponds to model parameters}
}


\newglossaryentry{Pseudo-Autosomal Region}
{
        name=Psuedo-Autosomal Region,
        description={a region of homology between mammalian X and Y chromosome undergoing one obligatory cross-over during meiosis}
}

\newglossaryentry{Substitution models}
{
        name={Substitution Model},plural={substitution models},
        text=substitution model,
        description={a Markov process that is used to describe how evolution happened along a edge}
}

\newglossaryentry{Maximum Likelihood}
{
        name=Maximum Likelihood,
        text=Maximum Likelihood,
        description={methods by which a model is ‘fit’ by returning the model parameters that produce the highest probability of generating the observed sequence data.}
}

\newglossaryentry{models}
{
        name={Model},plural={models},
        text={models},
        description={a collection of assumptions about the shape of a phylogenetic tree and the processes that pertain to each of the edges of a tree}
}

\newglossaryentry{nested}
{
        name=Nested,
        text=nested,
        description={One `null’ model is considered nested in an `alternate’ model if it can be specified simply by imposing restrictions on the parameters }
}

\newglossaryentry{df}
{
        name=df,
        text=df,
        description={ }
}

\newglossaryentry{$^5$mC}
{
        name=Cytosine methylated at the $5'$ carbon position,
        text=df,
        description={ }
}

%  asymptotic distribution